engine/src/test/java/org/camunda/bpm/engine/test/api/runtime/RestartProcessInstanceSyncTest.java

```diff
@@ -122,6 +123,10 @@ public void shouldRestartSimpleProcessInstance() {
     ProcessInstance restartedProcessInstance = runtimeService.createProcessInstanceQuery().active().singleResult();
     Task restartedTask = engineRule.getTaskService().createTaskQuery().processInstanceId(restartedProcessInstance.getId()).active().singleResult();
     Assert.assertEquals(task.getTaskDefinitionKey(), restartedTask.getTaskDefinitionKey());
+
+    HistoricProcessInstanceEntity historicProcessInstanceEntity = (HistoricProcessInstanceEntity) historyService.createHistoricProcessInstanceQuery().processInstanceId(restartedProcessInstance.getId()).singleResult();
```

Comment:
The test fails when commenting out several places where that contribute to setting the `RestartedProcessInstanceId` such as in the [StartProcessInstanceAtActivitiesCmd](https://github.com/camunda/camunda-bpm-platform/pull/4420/files#diff-38fa42d7a5cfae497358dee48627546c3dd1b7fba640a76007bb3904e99f50edR106), [StartProcessInstanceAtActivitiesCmd](https://github.com/camunda/camunda-bpm-platform/pull/4420/files#diff-92cf4f65d9df8ce5f4c22f4d38e5da16808221da652ed38b02f8970263e40711), [DefaultHistoryEventProducer](https://github.com/camunda/camunda-bpm-platform/pull/4420/files#diff-62ba4b3b7f0a7af63409e956be683887bf17f8f68b2668acecc5c6abb8ca770bR207) 👍